schema/cassandra/cadence/versioned/s0.0-0.23/schema_0_23.cql

```diff
@@ -0,0 +1,478 @@
+CREATE TYPE shard (
```

Comment:
How about breaking this file into two such as 1) Schema definitions 2) Data inserts? The major chunk of this file is the schema definition, and it could easily be generated by just dumping the output of `DESC KEYSPACE cadence` to a file. Inserts are only a few and very likely to remain unchanged over the course of several releases. If we follow this model, it'll be very easy to generate future squashes that go from v0.0 to vX.Y.

Reply:
Splitting the inserts and schema is a good idea, I'll check if the order of execution is preserved from the manifest, and will split if that's the case

On the `DESC KEYSPACE` - it was initially the idea for importing, however the documentation at https://docs.datastax.com/en/dse/5.1/cql/cql/cql_reference/cqlsh_commands/cqlshDescribeKeyspace.html says

> CAUTION: Verify all settings before executing the full output, some options may be cluster specific in the WITH statement.

and this is exactly what is appended, for example on docker instance

```
CREATE TABLE cadence.events (
    domain_id uuid,
    workflow_id text,
    run_id uuid,
    first_event_id bigint,
    data blob,
    data_encoding text,
    data_version int,
    event_batch_version bigint,
    range_id bigint,
    tx_id bigint,
    PRIMARY KEY ((domain_id, workflow_id, run_id), first_event_id)
) WITH CLUSTERING ORDER BY (first_event_id ASC)
    AND bloom_filter_fp_chance = 0.1
    AND caching = {'keys': 'ALL', 'rows_per_partition': 'NONE'}
    AND comment = ''
    AND compaction = {'class': 'org.apache.cassandra.db.compaction.LeveledCompactionStrategy'}
    AND compression = {'chunk_length_in_kb': '64', 'class': 'org.apache.cassandra.io.compress.LZ4Compressor'}
    AND crc_check_chance = 1.0
    AND dclocal_read_repair_chance = 0.1
    AND default_time_to_live = 0
    AND gc_grace_seconds = 864000
    AND max_index_interval = 2048
    AND memtable_flush_period_in_ms = 0
    AND min_index_interval = 128
    AND read_repair_chance = 0.0
    AND speculative_retry = '99PERCENTILE';
```

the cluster-specific settings do show up in here, so we either have to filter them out, or check the schema with and without shortcuts

Reply:
The `cql` file is split into schema and data in a511e75

Reply:
Awesome, and thanks for the additional context on `DESC KEYSPACE` behavior. That's definitely an important gotcha to watch out for.